add ADR for returning different states (#55)
govuk-one-login/account-management-stubs: docs/adrs/0001-different-states.md

Reviewer: Not sure if this was also discussed in our tech huddle but we could also make it more dynamic and allow it to receive the state from the payload itself? In that way we could test different scenarios on the fly (for demo purposes) in addition to the hard-coded scenarios.

Author: I did think about that, but there aren't many opportunities to change the payload.

Stripe for example, let you test different scenarios by changing the [credit card numbers](https://docs.stripe.com/testing#declined-payments).

So there are some text boxes that you could accept input and do something based on the input (like if the user put `wrong` as a password, you could simulate an incorrect password.

But most of the UI is you selecting from specific boxes, so how would that work?

# Returning different states in the API

## Context 

Currently, this stub API works quite basically; doing a very basic level of validation and responding with mostly static responses.

> As we develop One Login Home, there is a need to have a more flexibility in the API responses. For example, with MFA method management, we'd like to demonstrate users who have different combinations of MFA factors enabled, we can't do this with static responses.

Reviewer 2: ```suggestion
As we develop One Login Home, there is a need to have more flexibility in the API responses. For example, with MFA method management, we'd like to demonstrate users who have different combinations of MFA methods enabled, we can't do this with static responses.
```

## Option 1: Systems Manager Parameter Store

We could use the [systems manager parameter store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) to store a parameter that corresponds to a scenario. And then return the response we want.

> This would work in a similar way to how we use the secrets manager, in the lambda, we would read what the parameter is. The parameter could be changed either through the AWS console, or through an API. We could create a simple interface in the stub api to allow people to change the current scenario.

Reviewer 2: ```suggestion
This would work in a similar way to how we use the secrets manager currently.  In the lambda we would read read the parameter from secrets manager and use it to determine the response. The parameter could be changed either through the AWS console, or through an API. We could create a simple interface in the stub api to allow people to change the current scenario.
```
